repair/repair.cc

```diff
@@ -915,12 +921,25 @@ int repair_service::do_repair_start(sstring keyspace, std::unordered_map<sstring
         options.column_families.size() ? options.column_families : list_column_families(db.local(), keyspace);
     if (cfs.empty()) {
         rlogger.info("repair[{}]: completed successfully: no tables to repair", id.uuid());
-        return id.id;
+        return make_ready_future<int>(id.id);
     }
 
-    // Do it in the background.
-    (void)repair_tracker().run(id, [this, &db, id, keyspace = std::move(keyspace),
-            cfs = std::move(cfs), ranges = std::move(ranges), options = std::move(options), ignore_nodes = std::move(ignore_nodes)] () mutable {
+    auto it = options_map.find(repair_options::COLUMNFAMILIES_KEY);
+    sstring table = it == options_map.end() ? "" : it->second;
+    auto task_impl_ptr = std::make_unique<user_requested_repair_task_impl>(_repair_module, id, std::move(keyspace), std::move(table), format("{}", streaming::stream_reason::repair), "", std::move(cfs), std::move(ranges), std::move(options), std::move(ignore_nodes));
```

Comment:
The table is a incorrect abstraction for user_requested_repair_task_impl.

A user task should be able to repair multiple tables. Instead of a single table.